tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -7067,6 +7071,19 @@ void CGMSHLSLRuntime::EmitHLSLAggregateCopy(CodeGenFunction &CGF, llvm::Value *S
     SmallVector<Value *, 4> idxList;
     EmitHLSLAggregateCopy(CGF, SrcPtr, DestPtr, idxList, Ty, Ty, SrcPtr->getType());
 }
+
+// Make sure all element type of struct is same type.
+static bool IsStructWithSameElementType(llvm::StructType *ST, llvm::Type *Ty) {
```

Comment:
struct containing arrays of your type?

Reply:
Supported array inside struct with same type.